### PR TITLE
Move bench under `test` folder

### PR DESCRIPTION
**Pull request by HarelM**
## Launch Checklist

This PR moves `bench` folder under `test` where I believe it should be.
I've ran the bench locally but I'm not 100% sure it is working as expected.
@wipfli @birkskyum can you guys make sure I didn't break anything here?

 - [x] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [x] Briefly describe the changes in this PR.

### CODE REVIEW

**Comment by birkskyum** [comment]
It fits better there for sure. I upload-benchmarks.yml github workflow needs to get paths updated as well.

**Review by birkskyum — CHANGES_REQUESTED** [review]
LGTM. Only the paths in git workflow (upload-benchmarks.yml) left to update.

File: test/bench/README.md [hunk 1]
```diff
@@ -6,11 +6,11 @@ Benchmarks help us catch performance regressions and improve performance.
 
 Start the benchmark server with `npm run start-bench`.
 
-To run all benchmarks, open [the benchmark page, `http://localhost:9966/bench/versions`](http://localhost:9966/bench/versions).
+To run all benchmarks, open [the benchmark page, `http://localhost:9966/test/bench/versions`](http://localhost:9966/bench/versions).
```
Comment (wipfli) [inline test/bench/README.md:9]: http://localhost:9966/bench/versions -> http://localhost:9966/test/bench/versions

**Comment by wipfli** [comment]
Tested running benchmarks locally and they worked the same way as they used to. Please fix link in readme. Other than that, looks good to me.

**Comment by wipfli** [comment]
> When I opened the browser I saw some errors related to source_cache and some warning about a missing icon.
> Is this expected?

I don't know. Checkout main and compare how it was before.

**Review by birkskyum — CHANGES_REQUESTED** [review]
benchmarks-paths

Found these paths as well that I think needs attention

**Comment by wipfli** [comment]
Please also make sure that `npm run benchmark` completes all benchmarks successfully.

**Comment by HarelM** [comment]
I definitely don't like the backward compatibility need here. This is counter productive when talking about refactoring and moving files and other things around.
The fact that we use code that is not address agnostic is just bad practice.
I don't know how to properly solve this, but it's just another sign (besides the need to upload js files for every version) that this is poorly written... :-(
I'll open an issue about it...

**Comment by birkskyum** [comment]
I agree, it's really annoying... tripped me the other day with the es modules as well, and I'm thinking it might be more reasonable for us to upload the maplibre-gl.js build instead of the benchmarks build, because do we even need a such.

**Comment by birkskyum** [comment]
We have all the old versions here already: [https://cdn.skypack.dev/maplibre-gl@^1.3](https://cdn.skypack.dev/maplibre-gl@%5E1.3), so we could use the releases, and then the current branch, and forget about "main" in the comparisons...

**Comment by HarelM** [comment]
I added a start-server-at-test to fix the issue with the relative path, but now the css file is not available since dist folder is not part of this server. I can hack that too I guess, but I don't know if this is the right direction as I said before.
We also have a css from mapbox that we are loading as part of the benchmarks with I believe we should stop using:
`<link href='https://www.mapbox.com/base/latest/base.css' rel='stylesheet' />`
I'm not sure what it does, but it is used in the styles benchmark.

**Comment by birkskyum** [comment]
How about just using this url for the css? https://unpkg.com/maplibre-gl@2.1.1/dist/maplibre-gl.css

**Comment by HarelM** [comment]
Seems like the tests are running without loading the css, so I tend to say that it's redundant...
I need to see if removing the mapbox css is possible too.

**Comment by birkskyum** [comment]
Sure, the gl-stats.ts pull in the css from /dist/ , so that still works

**Comment by HarelM** [comment]
I don't see that gl-stat uses css, it pull the code and replaces that when running, but that's a different story.
In any case, I think this PR is read.
I'm not super exited about it, but it's better than what it used to be I believe.
I don't think it breaks anything as far as I tested.
Feel free to test this too as I'm not super familiar with the benchmarks usage...

**Comment by birkskyum** [comment]
Great! I'm almost through the whole benchmarks.

**Review by birkskyum — APPROVED** [review]
Same performance as main, so it lgtm!
benchmarks-cli

**Comment by xabbu42** [comment]
Sorry to be this late with the remark, but I think the benchmark documentation at https://github.com/maplibre/maplibre-gl-js/blob/main/test/bench/README.md still needs to be updated to match this changes.

**Pull request merged**